src/main/cc/proxy_client/javac_remote_actions.cc

```diff
@@ -0,0 +1,44 @@
+#include "src/main/cc/proxy_client/javac_remote_actions.h"
+
+#include <algorithm>
+
+#include "absl/strings/str_split.h"
+
+namespace remote_client {
+
+const int kMaxArgsToCheckForJavac = 8;
+
+const char* kJavacWrapperCommand = "javac_wrapper ";
+const char* kJavacCommand = "javac ";
+
+using std::min;
+
+void DetermineJavacActionInputs(int argc, char** argv, set<string>* inputs) {
+  for (int i = 0; i < argc; ++i) {
+    string cur(argv[i]);
+    if (cur[0] == '@') {
+      inputs->insert(cur.substr(1));
+    } else {
+      for (const auto& cmd : absl::StrSplit(cur, ' ', absl::SkipEmpty())) {
+        if (cmd[0] == '@') {
+          inputs->insert(string(cmd.substr(1)));
+        }
+      }
+    }
+  }
+}
+
+bool IsJavacAction(int argc, char** argv) {
+  const int num_args_to_check = min(kMaxArgsToCheckForJavac, argc);
```

Comment:
what is the purpose of this?

Reply:
Its an optimization that I saw in some other part of the code that Ola had :) Basically I only have to look at the first 8 args to determine if its a Javac action /not.